### ISSUE TITLE
段错误核心已转存

### ISSUE THREAD

**Issue by Martin8623** [geonhobang/RadarDistill#6]
I encountered a segment error during the process of reproducing the code, which caused the code to run interrupted. Of course, the code runs for a period of time and the error is not fixed each time.段错误

**Comment by Martin8623** [comment]
The error message when using GDB is[
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `python train.py --cfg_file cfgs/nuscenes_models/
Program terminated with signal SIGILL, Illegal instruction.
#0  0x00007fa18000dac2 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
[Current thread is 1 (Thread 0x7fa1e928a200 (LWP 10603))]
(gdb) bt
#0  0x00007fa18000dac2 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#1  0x00007fa17fe53090 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#2  0x00007fa17fe5314e in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#3  0x00007fa1801ad497 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#4  0x00007fa1801ad6dd in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#5  0x00007fa17ff41344 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#6  0x00007fa17fe13593 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#7  0x00007fa17fe13d5d in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#8  0x00007fa17fe1654a in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#9  0x00007fa17fff3efe in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so
#10 0x00007fa1de04c5ab in ?? () from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/../../../../libcudart.so.11.0
#11 0x00007fa1de09c93e in cudaLaunchKernel () from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/../../../../libcudart.so.11.0
#12 0x00007fa0e945d7d4 in __device_stub__ZN2at6native29vectorized_elementwise_kernelILi4EZZZNS0_16sqrt_kernel_cudaERNS_18TensorIteratorBaseEENKUlvE_clEvENKUlvE2_clEvEUnvhdl0_1_0_PFvS3_ENS0_16sqrt_kernel_cudaE2_ffENS_6detail5ArrayIPcLi2EEEEEviT0_T1_(int, __nv_hdl_wrapper_t<false, true, __nv_dl_tag<void (*)(at::TensorIteratorBase&), &at::native::sqrt_kernel_cuda, 2u>, float (float)>&, at::detail::Array<char*, 2>&) () from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/libtorch_cuda_cu.so
#13 0x00007fa0e948aea5 in void at::native::gpu_kernel_impl<__nv_hdl_wrapper_t<false, true, __nv_dl_tag<void (*)(at::TensorIteratorBase&), &at::native::sqrt_kernel_cuda, 2u>, float (float)> >(at::TensorIteratorBase&, __nv_hdl_wrapper_t<false, true, __nv_dl_tag<void (*)(at::TensorIteratorBase&), &at::native::sqrt_kernel_cuda, 2u>, float (float)> const&)
    () from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/libtorch_cuda_cu.so
#14 0x00007fa0e94b50f3 in void at::native::gpu_kernel<__nv_hdl_wrapper_t<false, true, __nv_dl_tag<void (*)(at::TensorIteratorBase&), &at::native::sqrt_kernel_cuda, 2u>, float (float)> >(at::TensorIteratorBase&, __nv_hdl_wrapper_t<false, true, __nv_dl_tag<void (*)(at::TensorIteratorBase&), &at::native::sqrt_kernel_cuda, 2u>, float (float)> const&) ()
   from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/libtorch_cuda_cu.so
#15 0x00007fa0e946b0ea in at::native::sqrt_kernel_cuda(at::TensorIteratorBase&) ()
   from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/libtorch_cuda_cu.so
#16 0x00007fa0e7d710e7 in at::(anonymous namespace)::wrapper_sqrt(at::Tensor const&) ()
   from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/libtorch_cuda_cu.so
#17 0x00007fa0e7d7119c in c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (at::Tensor const&), &at::(anonymous namespace)::wrapper_sqrt>, at::Tensor, c10::guts::typelist::typelist<at::Tensor const&> >, at::Tensor (at::Tensor const&)>::call(c10::OperatorKernel*, c10::DispatchKeySet, at::Tensor const&) () from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/libtorch_cuda_cu.so
#18 0x00007fa12e07d58f in at::Tensor c10::Dispatcher::redispatch<at::Tensor, at::Tensor const&>(c10::TypedOperatorHandle<at::Tensor (at::Tensor const&)> const&, c10::DispatchKeySet, at::Tensor const&) const [clone .isra.151] () from /home/m/anaconda3/envs/RadarDistill/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so
#19 0x00007fa12e07defe in at::_ops::sqrt::redispatch(c10::DispatchKeySet, at::Tensor const&) ()

**Comment by geonhobang** [comment]
I followed the environment described in GETTING_STARTED.md for RadarDistill, and my setup/validation was done on RTX 3090 (Ampere, CC 8.6). Please note that the exact CUDA stack can vary by GPU model and driver, so a driver–CUDA runtime–PyTorch CUDA build mismatch may cause errors like SIGILL in libcuda.so.

**Comment by Martin8623** [comment]
Thank you for your reply. I am using the same environment configuration as you and running on a 4090. The problem is quite strange. When I load the nuscenes-mini dataset, there are no errors. However, using the complete nuscenes dataset causes errors, and the error location is different each time, all resulting in the aforementioned error.I am using an RTX 4090 (8.9), which I believe may be the only difference between us. Similarly, I have run other dataset code models without encountering any errors, but running other code with the complete nuscenes dataset also causes the same issue.
2025-09-25 11:31:50,131   INFO  Train:    1/40 (  2%) [1799/5910 ( 30%)]  Loss: 93.70 (122.)  LR: 1.008e-04  Time cost: 13:52/57:45 [13:53/54:56:05]  Acc_iter 1800        Data time: 0.01(0.01)  Forward time: 0.58(0.84)  Batch time: 0.59(0.84)
2025-09-25 11:32:25,867   INFO  Train:    1/40 (  2%) [1849/5910 ( 31%)]  Loss: 119.6 (122.)  LR: 1.008e-04  Time cost: 14:28/56:38 [14:28/54:31:14]  Acc_iter 1850        Data time: 0.01(0.01)  Forward time: 0.51(0.83)  Batch time: 0.52(0.84)
2025-09-25 11:32:57,668   INFO  Save latest model to /home/m/RadarDistill/output/radar_distill/radar_distill_train/default/ckpt/latest_model
2025-09-25 11:33:01,176   INFO  Train:    1/40 (  2%) [1899/5910 ( 32%)]  Loss: 131.1 (122.)  LR: 1.009e-04  Time cost: 15:03/55:32 [15:04/54:07:04]  Acc_iter 1900        Data time: 0.00(0.01)  Forward time: 0.69(0.82)  Batch time: 0.69(0.83)
段错误 (核心已转储)

**Comment by Martin8623** [comment]
I am using an RTX 4090 (8.9), which I think might be the only difference between us.
